content/actions/using-workflows/workflow-commands-for-github-actions.md

```diff
@@ -368,6 +368,91 @@ jobs:
 
 {% endpowershell %}
 
+### Example: Masking a generated output within a single job
+
+> **Note**
+> The order matters here, you *must* use `add-mask` before you use `set-output`.
+
+As long as you don't need to pass your secret from one job to another job, you can:
+1. Generate the secret (without outputting it).
+2. Tag it with `add-mask`.
+3. Use `set-output` to make it available to other steps within your job.
+
+{% bash %}
+
+```yaml{:copy}
+on: push
+jobs:
+  generate-a-secret-output:
+    runs-on: ubuntu-latest
+    steps:
+      - id: sets-a-secret
+        name: Generate, mask, and output a secret
+        run: |
+          the_secret=$((RANDOM))
+          echo "::add-mask::$the_secret"
+          echo "::set-output name=secret-number::$the_secret"
+      - name: Use that secret output (protected by a mask)
+        run: |
+          echo "the secret number is ${{ steps.sets-a-secret.outputs.secret-number }}"
+```
+
+{% endbash %}
+
+### Scenario: Passing a secret between jobs or workflows
+
+#### Setup
+1. Set up a secret-store to be responsible for holding secrets (e.g. Vault).
+2. Generate a key for reading/writing to that secret-store.
+3. Add that [key as a secret to the repository](/actions/security-guides/encrypted-secrets).
+4. Create a workflow with:
+
+#### Workflow
+
+> **Note**
+> `secret-store` refers to an imaginary application with imaginary verbs (`store-secret` and `retrieve-secret`).
+> `some/secret-store@v1` would be responsible for installing that application and configuring it to connect to an `instance` with `credentials`.
```

Comment:
Otherwise, here's a wording suggestion to add a bit more context:

```suggestion
{% note %}

**Note**: This workflow uses an imaginary secret store, `secret-store`, which has imaginary commands `store-secret` and `retrieve-secret`. `some/secret-store@ 27b31702a0e7fc50959f5ad993c78deac1bdfc29` is an imaginary action that installs the `secret-store` application and configures it to connect to an `instance` with `credentials`.

{% endnote %}
```

Reply:
I don't have an actual example. We looked at Vault years ago and discarded it. This is the first time in ages where it seemed like a useful thing...